deep-learning/src/main/python/synapse/ml/dl/LitDeepTextModel.py

```diff
@@ -0,0 +1,151 @@
+    if _TRANSFORMERS_EQUAL_4_15_0:
+        from transformers import AutoModelForSequenceClassification
+    else:
+        raise RuntimeError(
+            "transformers should be == 4.15.0, found: {}".format(
+                transformers.__version__
+            )
+        )
+else:
+    raise ModuleNotFoundError("module not found: transformers")
+
+
+class LitDeepTextModel(pl.LightningModule):
+    def __init__(
+        self,
+        checkpoint,
+        text_col,
+        label_col,
+        num_labels,
+        additional_layers_to_train,
+        optimizer_name,
+        loss_name,
+        train_from_scratch=True,  # train from scratch gives the best performance
+    ):
+        super(LitDeepTextModel, self).__init__()
+
+        self.checkpoint = checkpoint
+        self.text_col = text_col
+        self.label_col = label_col
+        self.num_labels = num_labels
+        self.additional_layers_to_train = additional_layers_to_train
+        self.optimizer_name = optimizer_name
+        self.loss_name = loss_name
+        self.train_from_scratch = train_from_scratch
+
+        self._check_params()
+
+        self.save_hyperparameters(
+            "checkpoint",
+            "text_col",
+            "label_col",
+            "num_labels",
+            "additional_layers_to_train",
+            "optimizer_name",
+            "loss_name",
+            "train_from_scratch",
+        )
+
+    def _check_params(self):
+        # HUGGINGFACE.TRANSFORMERS
+        try:
+            self.model = AutoModelForSequenceClassification.from_pretrained(
+                self.checkpoint, num_labels=self.num_labels
+            )
+        except Exception as err:
+            raise ValueError(
+                f"No checkpoint {self.checkpoint} found: {err=}, {type(err)=}"
+            )
+
+        if self.loss_name.lower() not in F.__dict__:
+            raise ValueError("No loss function: {} found".format(self.loss_name))
+        self.loss_fn = F.__dict__[self.loss_name.lower()]
+
+        optimizers_mapping = {
+            key.lower(): value
+            for key, value in optim.__dict__.items()
+            if inspect.isclass(value) and issubclass(value, optim.Optimizer)
+        }
+        if self.optimizer_name.lower() not in optimizers_mapping:
+            raise ValueError("No optimizer: {} found".format(self.optimizer_name))
+        self.optimizer_fn = optimizers_mapping[self.optimizer_name.lower()]
+
+    def forward(self, **inputs):
+        return self.model(**inputs)
+
+    def configure_optimizers(self):
+        if not self.train_from_scratch:
+            # Freeze those weights
+            for p in self.model.base_model.parameters():
+                p.requires_grad = False
+            self._fine_tune_layers()
+        params_to_update = filter(lambda p: p.requires_grad, self.model.parameters())
+        # note that for different model recommended lr is different
+        return self.optimizer_fn(params_to_update, lr=5e-5)
```

Comment:
this seems low for linear probe learning, do you think thats why we couldnt get linear probe to train well? Also here and elsewhere we sholuld make lr a top level param as it is arguably most important knob to play with

Reply:
You're right, we should expose this param. This param has to be this low for bert models, we should 1. add this to param and 2. provide a default mapping of reasonable values if possible.